nomad/provider.go

```diff
@@ -69,10 +94,23 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	config.TLSConfig.ClientKey = d.Get("key_file").(string)
 	config.SecretID = d.Get("secret_id").(string)
 
+	// Get the vault token from the config, VAULT_TOKEN
+	// or ~/.vault-token (in that order)
+	var err error
+	vaultToken := d.Get("vault_token").(string)
+	if vaultToken == "" {
+		vaultToken, _ = getToken()
```

Comment:
Why is this error being ignored? If it's because we don't want to error out on a file that's optional, I get that. From the default implementation[1], it looks like it doesn't return an error if the file doesn't exist, so this should work as expected if no file is detected, but we could/should still return any error we run across, as it generally would indicate the user had settings, but we failed to retrieve/parse them. Which, as a user, I'd like to know about, rather than silently failing.